<a href="https://colab.research.google.com/github/ChiefRz/Proyek_Analisis_Data_AQI/blob/main/Proyek_Analisis_Data.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Proyek Analisis Data: AIR QUALITY DATASET
- **Nama:** RHEYZA AVTA NAUFALDY
- **Email:** rheyzaavtanaufaldy@gmail.com
- **ID Dicoding:** rheyzaan11

## Menentukan Pertanyaan Bisnis

- Bagaimana perubahan bulanan tingkat PM10 pada setiap stasiun ?
- Bagaimana kondisi tingkat PM10 antara musim dingin, semi, panas, dan gugur di setiap stasiun?

## Import Semua Packages/Library yang Digunakan

In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px
import os
import folium

Membuat file requirements.txt

In [3]:
!pip install pipreqsnb
!pipreqsnb --force

  Preparing metadata (setup.py) ... done
  Preparing metadata (setup.py) ... done
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 798.3/798.3 kB 21.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.6/1.6 MB 50.9 MB/s eta 0:00:00
  Created wheel for pipreqsnb: filename=pipreqsnb-0.2.4-py3-none-any.whl size=4129 sha256=01d0edcd199bfc03bcd97909e9c8c597d50eaa18d5612aeb645510e9c529a194
  Stored in directory: /root/.cache/pip/wheels/c3/1b/47/ca78a979321cb4bf9e812da5157fb754712f43d094bc60bf61
  Created wheel for docopt: filename=docopt-0.6.2-py2.py3-none-any.whl size=13706 sha256=3dec76f5e7f16c543f893e89796098f0938ff5b7f8ca296fb87de0b1ffeace99
  Stored in directory: /root/.cache/pip/wheels/fc/ab/d4/5da2067ac95b36618c629a5f93f809425700506f72c9732fac
Successfully built pipreqsnb docopt
  Attempting uninstall: ipython
    Found existing installation: ipython 7.34.0
    Uninstalling ipython-7.34.0:
      Successfully uninstalled ipython-7.34.0
ERROR: pip's dependency resolver does not 

pipreqs  --force /content
INFO: Not scanning for jupyter notebooks.
Please, verify manually the final list of requirements.txt to avoid possible dependency confusions.
Please, verify manually the final list of requirements.txt to avoid possible dependency confusions.
Please, verify manually the final list of requirements.txt to avoid possible dependency confusions.
Please, verify manually the final list of requirements.txt to avoid possible dependency confusions.
Please, verify manually the final list of requirements.txt to avoid possible dependency confusions.
INFO: Successfully saved requirements file in /content/requirements.txt


## Data Wrangling

### Gathering Data

Mendapatkan dan menyimpan daftar file yang ada dalam direktori

In [4]:
data_pencemaran_files = os.listdir('/content/drive/MyDrive/Colab Notebooks/Proyek Analisis Data/dataset/')
data_pencemaran_files

['PRSA_Data_Aotizhongxin_20130301-20170228.csv',
 'PRSA_Data_Wanshouxigong_20130301-20170228.csv',
 'PRSA_Data_Tiantan_20130301-20170228.csv',
 'PRSA_Data_Nongzhanguan_20130301-20170228.csv',
 'PRSA_Data_Changping_20130301-20170228.csv',
 'PRSA_Data_Wanliu_20130301-20170228.csv',
 'PRSA_Data_Dongsi_20130301-20170228.csv',
 'PRSA_Data_Gucheng_20130301-20170228.csv',
 'PRSA_Data_Dingling_20130301-20170228.csv',
 'PRSA_Data_Huairou_20130301-20170228.csv',
 'PRSA_Data_Guanyuan_20130301-20170228.csv',
 'PRSA_Data_Shunyi_20130301-20170228.csv']

Inisialisasi dataframe kosong

In [5]:
data_pencemaran = pd.DataFrame()

Mengumpulkan dan menggabungkan data dari beberapa file CSV ke dalam satu DataFrame

In [6]:
for file in data_pencemaran_files:
    data = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Proyek Analisis Data/dataset/' + file)
    data_pencemaran = pd.concat([data_pencemaran, data], ignore_index=True)

data_pencemaran

No  year  month  day  hour  PM2.5  PM10   SO2   NO2      CO    O3  \
0           1  2013      3    1     0    4.0   4.0   4.0   7.0   300.0  77.0   
1           2  2013      3    1     1    8.0   8.0   4.0   7.0   300.0  77.0   
2           3  2013      3    1     2    7.0   7.0   5.0  10.0   300.0  73.0   
3           4  2013      3    1     3    6.0   6.0  11.0  11.0   300.0  72.0   
4           5  2013      3    1     4    3.0   3.0  12.0  12.0   300.0  72.0   
...       ...   ...    ...  ...   ...    ...   ...   ...   ...     ...   ...   
420763  35060  2017      2   28    19   27.0  72.0   8.0  92.0   800.0  16.0   
420764  35061  2017      2   28    20   47.0  55.0  17.0  86.0  1100.0  19.0   
420765  35062  2017      2   28    21   18.0  28.0   4.0  30.0   500.0  64.0   
420766  35063  2017      2   28    22   18.0  20.0   9.0  33.0   500.0  59.0   
420767  35064  2017      2   28    23   15.0  22.0  13.0  34.0   500.0  60.0   

        TEMP    PRES  DEWP  RAIN   wd  WSPM       station  
0       -0.7  1023.0 -18.8   0.0  NNW   4.4  Aotizhongxin  
1       -1.1  1023.2 -18.2   0.0    N   4.7  Aotizhongxin  
2       -1.1  1023.5 -18.2   0.0  NNW   5.6  Aotizhongxin  
3       -1.4  1024.5 -19.4   0.0   NW   3.1  Aotizhongxin  
4       -2.0  1025.2 -19.5   0.0    N   2.0  Aotizhongxin  
...      ...     ...   ...   ...  ...   ...           ...  
420763  10.3  1014.2 -12.4   0.0    W   1.8        Shunyi  
420764   9.8  1014.5  -9.9   0.0   NW   1.5        Shunyi  
420765   9.1  1014.6 -12.7   0.0   NE   1.7        Shunyi  
420766   7.1  1015.2 -13.2   0.0  WNW   1.8        Shunyi  
420767   7.4  1014.9 -11.9   0.0    N   1.4        Shunyi  

[420768 rows x 18 columns]

**Insight:**
- Diketahui terdapat 18 kolom dalam dataFrame data_pencemaran
- Diketahui data pada tahun 2013 hanya tersedia mulai bulan ke-3
- Diketahui data pada tahun 2017 hanya sampai bulan ke-2
- Kolom Month adalah kolom yang berisi data-data bulan dalam angka
- Kolom Day adalah kolom yang berisi data-data hari dalam angka
- Kolom Hour adalah adalah kolom yang berisi data-data jam dalam angka
- Kolom PM2.5 adalah kolom yang berisi data-data particulate matter 2.5 mikrometer
- Kolom PM10 adalah kolom yang berisi data-data particulate matter 10 mikrometer
- Kolom SO2 adalah kolom yang berisi data-data Sulfur Dioxide
- Kolom NO2 adalah kolom yang berisi data-data Nitrogen Dioxide
- Kolom CO adalah kolom yang berisi data-data Carbon Monoxide
- Kolom O3 adalah kolom yang berisi data-data Ozone
- Kolom TEMP adalah kolom yang berisi data-data Temperatur
- Kolom PRES adalah kolom yang berisi data-data Pressure atau Tekanan Udara
- Kolom DEWP adalah kolom yang berisi data-data Dew Point atau Titik Embun
- Kolom RAIN adalah kolom yang berisi data-data Curah Hujan
- Kolom wd adalah kolom yang berisi data-data Wind Direction atau Arah Angin
- Kolom WSPM adalah kolom yang berisi data-data Wind Speed atau Kecepatan Angin
- Kolom station adalah kolom yang berisi data-data nama stasiun pengamatan kualitas udara


### Assessing Data

#### Menilai DataFrame data_pencemaran

Menampilkan informasi tentang data_pencemaran

In [7]:
data_pencemaran.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 420768 entries, 0 to 420767
Data columns (total 18 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   No       420768 non-null  int64  
 1   year     420768 non-null  int64  
 2   month    420768 non-null  int64  
 3   day      420768 non-null  int64  
 4   hour     420768 non-null  int64  
 5   PM2.5    412029 non-null  float64
 6   PM10     414319 non-null  float64
 7   SO2      411747 non-null  float64
 8   NO2      408652 non-null  float64
 9   CO       400067 non-null  float64
 10  O3       407491 non-null  float64
 11  TEMP     420370 non-null  float64
 12  PRES     420375 non-null  float64
 13  DEWP     420365 non-null  float64
 14  RAIN     420378 non-null  float64
 15  wd       418946 non-null  object 
 16  WSPM     420450 non-null  float64
 17  station  420768 non-null  object 
dtypes: float64(11), int64(5), object(2)
memory usage: 57.8+ MB


Menghitung jumlah nilai yang hilang dalam setiap kolom dari data_pencemaran

In [8]:
data_pencemaran.isna().sum()

No             0
year           0
month          0
day            0
hour           0
PM2.5       8739
PM10        6449
SO2         9021
NO2        12116
CO         20701
O3         13277
TEMP         398
PRES         393
DEWP         403
RAIN         390
wd          1822
WSPM         318
station        0
dtype: int64

Menghitung jumlah baris duplikat dalam data_pencemaran

In [9]:
print("Jumlah Duplikasi : ", data_pencemaran.duplicated().sum())

Jumlah Duplikasi :  0


Menampilkan gambaran umum dari data dalam data_pencemaran

In [10]:
data_pencemaran.describe()

No           year          month            day  \
count  420768.000000  420768.000000  420768.000000  420768.000000   
mean    17532.500000    2014.662560       6.522930      15.729637   
std     10122.116943       1.177198       3.448707       8.800102   
min         1.000000    2013.000000       1.000000       1.000000   
25%      8766.750000    2014.000000       4.000000       8.000000   
50%     17532.500000    2015.000000       7.000000      16.000000   
75%     26298.250000    2016.000000      10.000000      23.000000   
max     35064.000000    2017.000000      12.000000      31.000000   

                hour          PM2.5           PM10            SO2  \
count  420768.000000  412029.000000  414319.000000  411747.000000   
mean       11.500000      79.793428     104.602618      15.830835   
std         6.922195      80.822391      91.772426      21.650603   
min         0.000000       2.000000       2.000000       0.285600   
25%         5.750000      20.000000      36.000000       3.000000   
50%        11.500000      55.000000      82.000000       7.000000   
75%        17.250000     111.000000     145.000000      20.000000   
max        23.000000     999.000000     999.000000     500.000000   

                 NO2             CO             O3           TEMP  \
count  408652.000000  400067.000000  407491.000000  420370.000000   
mean       50.638586    1230.766454      57.372271      13.538976   
std        35.127912    1160.182716      56.661607      11.436139   
min         1.026500     100.000000       0.214200     -19.900000   
25%        23.000000     500.000000      11.000000       3.100000   
50%        43.000000     900.000000      45.000000      14.500000   
75%        71.000000    1500.000000      82.000000      23.300000   
max       290.000000   10000.000000    1071.000000      41.600000   

                PRES           DEWP           RAIN           WSPM  
count  420375.000000  420365.000000  420378.000000  420450.000000  
mean     1010.746982       2.490822       0.064476       1.729711  
std        10.474055      13.793847       0.821004       1.246386  
min       982.400000     -43.400000       0.000000       0.000000  
25%      1002.300000      -8.900000       0.000000       0.900000  
50%      1010.400000       3.100000       0.000000       1.400000  
75%      1019.000000      15.100000       0.000000       2.200000  
max      1042.800000      29.100000      72.500000      13.200000

**Insight:**
- Diketahui terdapat 5 Kolom bertipe integer
- Diketahui terdapat 11 Kolom bertipe float
- Diketahui terdapat 2 kolom bertipe object
- Diketahui hanya 6 dari 18 kolom yang tidak memiliki nilai yang hilang (misssing value)
- Kolom PM10 memiliki jumlah nilai yang hilang (missing value) sebanyak 6449 baris
- Diketahui tidak terdapat duplikasi di DataFrame data_pencemaran

### Cleaning Data

#### Membersihkan Tabel data_pencemaran

Melakukan Pengecekan Missing Value

In [11]:
data_pencemaran.isna().sum()

No             0
year           0
month          0
day            0
hour           0
PM2.5       8739
PM10        6449
SO2         9021
NO2        12116
CO         20701
O3         13277
TEMP         398
PRES         393
DEWP         403
RAIN         390
wd          1822
WSPM         318
station        0
dtype: int64

Menghapus missing value pada kolom PM10

In [12]:
data_pencemaran.dropna(subset=["PM10"], inplace=True)

Menghapus baris yang memiliki value year 2017

In [13]:
data_pencemaran = data_pencemaran[data_pencemaran['year'] != 2017]

**Insight:**
- Missing Value pada kolom PM10 sudah dibersihkan
- Data yang akan digunakan adalah data selain yang berisi 2017 pada kolom year

## Exploratory Data Analysis (EDA)

### Explore Tabel data_pencemaran

Konversi tahun ke variabel kategorikal

In [14]:
data_pencemaran.loc[:, 'month'] = data_pencemaran['month'].astype('category')
data_pencemaran.loc[:, 'year'] = data_pencemaran['year'].astype('category')
data_pencemaran.info()
print(data_pencemaran['month'].unique())

<class 'pandas.core.frame.DataFrame'>
Index: 397554 entries, 0 to 419351
Data columns (total 18 columns):
 #   Column   Non-Null Count   Dtype  
---  ------   --------------   -----  
 0   No       397554 non-null  int64  
 1   year     397554 non-null  int64  
 2   month    397554 non-null  int64  
 3   day      397554 non-null  int64  
 4   hour     397554 non-null  int64  
 5   PM2.5    395090 non-null  float64
 6   PM10     397554 non-null  float64
 7   SO2      393782 non-null  float64
 8   NO2      390743 non-null  float64
 9   CO       382386 non-null  float64
 10  O3       389353 non-null  float64
 11  TEMP     397292 non-null  float64
 12  PRES     397291 non-null  float64
 13  DEWP     397287 non-null  float64
 14  RAIN     397295 non-null  float64
 15  wd       396187 non-null  object 
 16  WSPM     397319 non-null  float64
 17  station  397554 non-null  object 
dtypes: float64(11), int64(5), object(2)
memory usage: 57.6+ MB
[ 3  4  5  6  7  8  9 10 11 12  1  2]


Grouping dan menghitung rata-rata TEMP

In [15]:
data_grouped = data_pencemaran.groupby(['station', 'month', 'year'])['PM10'].mean().reset_index()
data_grouped

station  month  year        PM10
0     Aotizhongxin      1  2014  125.670732
1     Aotizhongxin      1  2015  114.964179
2     Aotizhongxin      1  2016   86.884511
3     Aotizhongxin      2  2014  150.735069
4     Aotizhongxin      2  2015  123.586207
..             ...    ...   ...         ...
547  Wanshouxigong     11  2016  136.377778
548  Wanshouxigong     12  2013  121.998654
549  Wanshouxigong     12  2014  111.129363
550  Wanshouxigong     12  2015  180.304469
551  Wanshouxigong     12  2016  169.460189

[552 rows x 4 columns]

Membuat Fungsi untuk menentukan musim berdasarkan bulan

In [16]:
def menentukan_musim(month):
    if month in [12, 1, 2]:
        return 'Musim Dingin'
    elif month in [3, 4, 5]:
        return 'Musim Semi'
    elif month in [6, 7, 8]:
        return 'Musim Panas'
    elif month in [9, 10, 11]:
        return 'Musim Gugur'
    else:
        return 'Invalid month'

Merapkan fungsi untuk menambahkan kolom musim

In [17]:
data_grouped['musim'] = data_grouped['month'].apply(menentukan_musim)

data_grouped

station  month  year        PM10         musim
0     Aotizhongxin      1  2014  125.670732  Musim Dingin
1     Aotizhongxin      1  2015  114.964179  Musim Dingin
2     Aotizhongxin      1  2016   86.884511  Musim Dingin
3     Aotizhongxin      2  2014  150.735069  Musim Dingin
4     Aotizhongxin      2  2015  123.586207  Musim Dingin
..             ...    ...   ...         ...           ...
547  Wanshouxigong     11  2016  136.377778   Musim Gugur
548  Wanshouxigong     12  2013  121.998654  Musim Dingin
549  Wanshouxigong     12  2014  111.129363  Musim Dingin
550  Wanshouxigong     12  2015  180.304469  Musim Dingin
551  Wanshouxigong     12  2016  169.460189  Musim Dingin

[552 rows x 5 columns]

Membuat fungsi untuk mengkategorikan AQI

In [18]:
def calculate_aqi_pm10(concentration):
    BP_low = 55
    BP_high = 154
    I_low = 51
    I_high = 100

    AQI_PM10 = ((concentration - BP_low) / (BP_high - BP_low)) * (I_high - I_low) + I_low
    return AQI_PM10

Menghitung AQI untuk setiap nilai di kolom 'pm10' dan menambahkannya ke kolom baru 'aqi_pm10'

In [19]:
data_grouped['aqi_pm10'] = data_grouped['PM10'].apply(calculate_aqi_pm10)

print(data_grouped)

           station  month  year        PM10         musim    aqi_pm10
0     Aotizhongxin      1  2014  125.670732  Musim Dingin   85.978443
1     Aotizhongxin      1  2015  114.964179  Musim Dingin   80.679240
2     Aotizhongxin      1  2016   86.884511  Musim Dingin   66.781223
3     Aotizhongxin      2  2014  150.735069  Musim Dingin   98.384024
4     Aotizhongxin      2  2015  123.586207  Musim Dingin   84.946708
..             ...    ...   ...         ...           ...         ...
547  Wanshouxigong     11  2016  136.377778   Musim Gugur   91.277890
548  Wanshouxigong     12  2013  121.998654  Musim Dingin   84.160950
549  Wanshouxigong     12  2014  111.129363  Musim Dingin   78.781200
550  Wanshouxigong     12  2015  180.304469  Musim Dingin  113.019384
551  Wanshouxigong     12  2016  169.460189  Musim Dingin  107.652013

[552 rows x 6 columns]


Grouping dan menghitung rata-rata TEMP

In [20]:
data_grouped_aqi = data_grouped
data_grouped_aqi = data_grouped_aqi.groupby(['station', 'year'])['aqi_pm10'].mean().reset_index()
data_grouped_aqi

station  year   aqi_pm10
0    Aotizhongxin  2013  79.655928
1    Aotizhongxin  2014  84.033077
2    Aotizhongxin  2015  79.262817
3    Aotizhongxin  2016  70.505994
4       Changping  2013  68.919711
5       Changping  2014  76.751244
6       Changping  2015  70.164366
7       Changping  2016  66.395176
8        Dingling  2013  62.747008
9        Dingling  2014  68.786120
10       Dingling  2015  66.112162
11       Dingling  2016  63.455943
12         Dongsi  2013  75.629594
13         Dongsi  2014  81.799801
14         Dongsi  2015  79.196621
15         Dongsi  2016  74.456523
16       Guanyuan  2013  77.095171
17       Guanyuan  2014  81.222532
18       Guanyuan  2015  76.423646
19       Guanyuan  2016  75.449343
20        Gucheng  2013  79.479608
21        Gucheng  2014  90.143810
22        Gucheng  2015  81.349313
23        Gucheng  2016  78.468812
24        Huairou  2013  71.202368
25        Huairou  2014  73.295830
26        Huairou  2015  68.070483
27        Huairou  2016  64.391575
28   Nongzhanguan  2013  75.157394
29   Nongzhanguan  2014  86.013504
30   Nongzhanguan  2015  77.024836
31   Nongzhanguan  2016  71.720666
32         Shunyi  2013  70.594343
33         Shunyi  2014  78.830288
34         Shunyi  2015  73.211100
35         Shunyi  2016  66.904821
36        Tiantan  2013  77.367569
37        Tiantan  2014  80.559271
38        Tiantan  2015  74.748370
39        Tiantan  2016  71.725261
40         Wanliu  2013  79.339980
41         Wanliu  2014  88.374769
42         Wanliu  2015  76.666635
43         Wanliu  2016  69.621214
44  Wanshouxigong  2013  78.856432
45  Wanshouxigong  2014  86.381655
46  Wanshouxigong  2015  78.784250
47  Wanshouxigong  2016  72.516236

Mengelompokkan AQI

In [21]:
def air_quality(aqi):
    if aqi <= 50:
        return 'Baik'
    elif aqi <= 100:
        return 'Sedang'
    elif aqi <= 150:
        return 'Tidak Sehat'
    elif aqi <= 200:
        return 'Sangat Tidak Sehat'
    else:
        return 'Berbahaya'

data_grouped_aqi['kualitas_udara'] = data_grouped_aqi['aqi_pm10'].apply(air_quality)
print(data_grouped_aqi)

          station  year   aqi_pm10 kualitas_udara
0    Aotizhongxin  2013  79.655928         Sedang
1    Aotizhongxin  2014  84.033077         Sedang
2    Aotizhongxin  2015  79.262817         Sedang
3    Aotizhongxin  2016  70.505994         Sedang
4       Changping  2013  68.919711         Sedang
5       Changping  2014  76.751244         Sedang
6       Changping  2015  70.164366         Sedang
7       Changping  2016  66.395176         Sedang
8        Dingling  2013  62.747008         Sedang
9        Dingling  2014  68.786120         Sedang
10       Dingling  2015  66.112162         Sedang
11       Dingling  2016  63.455943         Sedang
12         Dongsi  2013  75.629594         Sedang
13         Dongsi  2014  81.799801         Sedang
14         Dongsi  2015  79.196621         Sedang
15         Dongsi  2016  74.456523         Sedang
16       Guanyuan  2013  77.095171         Sedang
17       Guanyuan  2014  81.222532         Sedang
18       Guanyuan  2015  76.423646         Sedang


Menginisiasi data latitude dan longitude

In [22]:
data = {
    'station': ['Aotizhongxin', 'Changping', 'Dingling', 'Dongsi', 'Guanyuan', 'Gucheng', 'Huairou', 'Nongzhanguan', 'Shunyi', 'Tiantan', 'Wanliu', 'Wanshouxigong'],
    'latitude': [39.9092, 40.2000, 40.2350, 39.9330, 39.9510, 39.9130, 40.4300, 39.9320, 40.1300, 39.8840, 39.9790, 39.9050],
    'longitude': [116.3972, 116.2000, 116.1960, 116.4160, 116.3090, 116.2040, 116.6400, 116.4500, 116.6520, 116.4140, 116.3070, 116.3070]
}

df = pd.DataFrame(data)
df

station  latitude  longitude
0    Aotizhongxin   39.9092   116.3972
1       Changping   40.2000   116.2000
2        Dingling   40.2350   116.1960
3          Dongsi   39.9330   116.4160
4        Guanyuan   39.9510   116.3090
5         Gucheng   39.9130   116.2040
6         Huairou   40.4300   116.6400
7    Nongzhanguan   39.9320   116.4500
8          Shunyi   40.1300   116.6520
9         Tiantan   39.8840   116.4140
10         Wanliu   39.9790   116.3070
11  Wanshouxigong   39.9050   116.3070

Menggabungkan DataFrame df dan data_grouped_aqi

In [23]:
AQI_df = pd.merge(df, data_grouped_aqi, on='station', how='outer')
AQI_df

station  latitude  longitude  year   aqi_pm10 kualitas_udara
0    Aotizhongxin   39.9092   116.3972  2013  79.655928         Sedang
1    Aotizhongxin   39.9092   116.3972  2014  84.033077         Sedang
2    Aotizhongxin   39.9092   116.3972  2015  79.262817         Sedang
3    Aotizhongxin   39.9092   116.3972  2016  70.505994         Sedang
4       Changping   40.2000   116.2000  2013  68.919711         Sedang
5       Changping   40.2000   116.2000  2014  76.751244         Sedang
6       Changping   40.2000   116.2000  2015  70.164366         Sedang
7       Changping   40.2000   116.2000  2016  66.395176         Sedang
8        Dingling   40.2350   116.1960  2013  62.747008         Sedang
9        Dingling   40.2350   116.1960  2014  68.786120         Sedang
10       Dingling   40.2350   116.1960  2015  66.112162         Sedang
11       Dingling   40.2350   116.1960  2016  63.455943         Sedang
12         Dongsi   39.9330   116.4160  2013  75.629594         Sedang
13         Dongsi   39.9330   116.4160  2014  81.799801         Sedang
14         Dongsi   39.9330   116.4160  2015  79.196621         Sedang
15         Dongsi   39.9330   116.4160  2016  74.456523         Sedang
16       Guanyuan   39.9510   116.3090  2013  77.095171         Sedang
17       Guanyuan   39.9510   116.3090  2014  81.222532         Sedang
18       Guanyuan   39.9510   116.3090  2015  76.423646         Sedang
19       Guanyuan   39.9510   116.3090  2016  75.449343         Sedang
20        Gucheng   39.9130   116.2040  2013  79.479608         Sedang
21        Gucheng   39.9130   116.2040  2014  90.143810         Sedang
22        Gucheng   39.9130   116.2040  2015  81.349313         Sedang
23        Gucheng   39.9130   116.2040  2016  78.468812         Sedang
24        Huairou   40.4300   116.6400  2013  71.202368         Sedang
25        Huairou   40.4300   116.6400  2014  73.295830         Sedang
26        Huairou   40.4300   116.6400  2015  68.070483         Sedang
27        Huairou   40.4300   116.6400  2016  64.391575         Sedang
28   Nongzhanguan   39.9320   116.4500  2013  75.157394         Sedang
29   Nongzhanguan   39.9320   116.4500  2014  86.013504         Sedang
30   Nongzhanguan   39.9320   116.4500  2015  77.024836         Sedang
31   Nongzhanguan   39.9320   116.4500  2016  71.720666         Sedang
32         Shunyi   40.1300   116.6520  2013  70.594343         Sedang
33         Shunyi   40.1300   116.6520  2014  78.830288         Sedang
34         Shunyi   40.1300   116.6520  2015  73.211100         Sedang
35         Shunyi   40.1300   116.6520  2016  66.904821         Sedang
36        Tiantan   39.8840   116.4140  2013  77.367569         Sedang
37        Tiantan   39.8840   116.4140  2014  80.559271         Sedang
38        Tiantan   39.8840   116.4140  2015  74.748370         Sedang
39        Tiantan   39.8840   116.4140  2016  71.725261         Sedang
40         Wanliu   39.9790   116.3070  2013  79.339980         Sedang
41         Wanliu   39.9790   116.3070  2014  88.374769         Sedang
42         Wanliu   39.9790   116.3070  2015  76.666635         Sedang
43         Wanliu   39.9790   116.3070  2016  69.621214         Sedang
44  Wanshouxigong   39.9050   116.3070  2013  78.856432         Sedang
45  Wanshouxigong   39.9050   116.3070  2014  86.381655         Sedang
46  Wanshouxigong   39.9050   116.3070  2015  78.784250         Sedang
47  Wanshouxigong   39.9050   116.3070  2016  72.516236         Sedang

Menggabungkan DataFrame AQI_df dan data_grouped

In [24]:
AQI_all_df = pd.concat([AQI_df, data_grouped])
AQI_all_df

station  latitude  longitude  year    aqi_pm10 kualitas_udara  \
0     Aotizhongxin   39.9092   116.3972  2013   79.655928         Sedang   
1     Aotizhongxin   39.9092   116.3972  2014   84.033077         Sedang   
2     Aotizhongxin   39.9092   116.3972  2015   79.262817         Sedang   
3     Aotizhongxin   39.9092   116.3972  2016   70.505994         Sedang   
4        Changping   40.2000   116.2000  2013   68.919711         Sedang   
..             ...       ...        ...   ...         ...            ...   
547  Wanshouxigong       NaN        NaN  2016   91.277890            NaN   
548  Wanshouxigong       NaN        NaN  2013   84.160950            NaN   
549  Wanshouxigong       NaN        NaN  2014   78.781200            NaN   
550  Wanshouxigong       NaN        NaN  2015  113.019384            NaN   
551  Wanshouxigong       NaN        NaN  2016  107.652013            NaN   

     month        PM10         musim  
0      NaN         NaN           NaN  
1      NaN         NaN           NaN  
2      NaN         NaN           NaN  
3      NaN         NaN           NaN  
4      NaN         NaN           NaN  
..     ...         ...           ...  
547   11.0  136.377778   Musim Gugur  
548   12.0  121.998654  Musim Dingin  
549   12.0  111.129363  Musim Dingin  
550   12.0  180.304469  Musim Dingin  
551   12.0  169.460189  Musim Dingin  

[600 rows x 9 columns]

Menyimpan DataFrame AQI_all_df ke dalam file CSV

In [25]:
AQI_all_df.to_csv("/content/drive/MyDrive/Colab Notebooks/Proyek Analisis Data/all_data/AQI_all_df.csv")

**Insight:**
- Diketahui kolom yang akan dianalisis yaitu kolom stasiun, bulan, dan tahun, dan rata-rata konsentrasi PM10
- Diketahui Fungsi menentukan_musim digunakan untuk mengkategorikan bulan ke dalam musim yang berbeda (Dingin, Semi, Panas, Gugur)
- Diketahui AQI untuk PM10 dihitung dengan menggunakan rumus yang melibatkan batasan konsentrasi.
- Diketahui pengklasifikasian kualitas udara ke dalam kategori seperti 'Baik', 'Sedang', 'Tidak Sehat'
- Diketahui data dikelompokkan lagi berdasarkan stasiun dan tahun untuk mendapatkan rata-rata AQI pertahun untuk memberikan gambaran umum tentang kualitas udara di masing-masing stasiun.
- Diketahui Data stasiun yang mencakup informasi geospasial (latitude dan longitude)
- Diketahui Data stasiun digabungkan dengan data AQI yang memungkinkan visualisasi dan analisis lebih lanjut tentang distribusi kualitas udara di peta
- Diketahui AQI_all_df disimpan dalam format CSV
- Diketahui dengan data tersebut, dapat digunakan untuk membuat visualisasi yang menarik, seperti peta sebaran kualitas udara, grafik tren kualitas udara dari waktu ke waktu, dan lainnya.

## Visualization & Explanatory Analysis

### Pertanyaan 1: Bagaimana perubahan bulanan tingkat PM10 pada setiap stasiun ?


Menghapus baris-baris dari AQI_all_df yang memiliki nilai NaN pada kolom month, PM10, dan musim

In [26]:
data_grouped = AQI_all_df.dropna(subset=['month', 'PM10', 'musim'])
data_grouped

station  latitude  longitude  year    aqi_pm10 kualitas_udara  \
0     Aotizhongxin       NaN        NaN  2014   85.978443            NaN   
1     Aotizhongxin       NaN        NaN  2015   80.679240            NaN   
2     Aotizhongxin       NaN        NaN  2016   66.781223            NaN   
3     Aotizhongxin       NaN        NaN  2014   98.384024            NaN   
4     Aotizhongxin       NaN        NaN  2015   84.946708            NaN   
..             ...       ...        ...   ...         ...            ...   
547  Wanshouxigong       NaN        NaN  2016   91.277890            NaN   
548  Wanshouxigong       NaN        NaN  2013   84.160950            NaN   
549  Wanshouxigong       NaN        NaN  2014   78.781200            NaN   
550  Wanshouxigong       NaN        NaN  2015  113.019384            NaN   
551  Wanshouxigong       NaN        NaN  2016  107.652013            NaN   

     month        PM10         musim  
0      1.0  125.670732  Musim Dingin  
1      1.0  114.964179  Musim Dingin  
2      1.0   86.884511  Musim Dingin  
3      2.0  150.735069  Musim Dingin  
4      2.0  123.586207  Musim Dingin  
..     ...         ...           ...  
547   11.0  136.377778   Musim Gugur  
548   12.0  121.998654  Musim Dingin  
549   12.0  111.129363  Musim Dingin  
550   12.0  180.304469  Musim Dingin  
551   12.0  169.460189  Musim Dingin  

[552 rows x 9 columns]

Menentukan Stasiun dan Tahun yang ingin di highlight

In [27]:
station_name1 = 'Aotizhongxin'
year1 = '2013'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [28]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [29]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu



In [30]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [31]:
station_name1 = 'Changping'
year1 = '2014'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [32]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [33]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu



In [34]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [35]:
station_name1 = 'Dingling'
year1 = '2015'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [36]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [37]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [38]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [39]:
station_name1 = 'Dongsi'
year1 = '2016'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [40]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [41]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [42]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [43]:
station_name1 = 'Guanyuan'
year1 = '2013'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [44]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [45]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [46]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [47]:
station_name1 = 'Gucheng'
year1 = '2014'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [48]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [49]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [50]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [51]:
station_name1 = 'Huairou'
year1 = '2015'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [52]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [53]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [54]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [55]:
station_name1 = 'Nongzhanguan'
year1 = '2016'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [56]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [57]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [58]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [59]:
station_name1 = 'Shunyi'
year1 = '2013'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [60]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [61]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [62]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [63]:
station_name1 = 'Tiantan'
year1 = '2014'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [64]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [65]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [66]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [67]:
station_name1 = 'Wanliu'
year1 = '2015'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [68]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [69]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [70]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [71]:
station_name1 = 'Wanshouxigong'
year1 = '2016'

Memvisualisasikan data PM10 dari stasiun tertentu antar tahun

In [72]:
rata_rata_pm10 = data_grouped[data_grouped['station'] == station_name1]
fig = px.line(rata_rata_pm10, x='month', y='PM10', color='year',
              color_discrete_sequence=px.colors.qualitative.Set3)

Membuat garis untuk tahun dipilih lebih tebal dibandingkan dengan garis untuk tahun-tahun lainnya

In [73]:
thick_year = year1
thick_line_width = 5
default_line_width = 2

for trace in fig.data:
    if trace.name == thick_year:
        trace.line.width = thick_line_width
    else:
        trace.line.width = default_line_width

Menyempurnakan dan menampilkan visualisasi grafik dengan pengaturan yang tertentu

In [74]:
fig.update_xaxes(type='category')
fig.update_xaxes(ticktext=rata_rata_pm10['month'].unique(), tickvals=rata_rata_pm10['month'].unique())
fig.update_layout(title=f'Rata-Rata PM10 setiap bulan di stasiun {station_name1} pada tahun {year1}',
                  yaxis_title='Rata-Rata PM10',
                  xaxis_title='Bulan',
                  legend_title='Tahun')

fig.show()

### Pertanyaan 2: Bagaimana kondisi tingkat PM10 antara musim dingin, semi, panas, dan gugur di setiap stasiun?

Menentukan Stasiun dan Tahun yang ingin di highlight

In [75]:
station_name2 = 'Aotizhongxin'
year2 = 2013

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [76]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [77]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [78]:
station_name2 = 'Changping'
year2 = 2014

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [79]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [80]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [81]:
station_name2 = 'Dingling'
year2 = 2015

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [82]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [83]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [84]:
station_name2 = 'Dongsi'
year2 = 2016

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [85]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [86]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [87]:
station_name2 = 'Guanyuan'
year2 = 2013

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [88]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [89]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [90]:
station_name2 = 'Gucheng'
year2 = 2014

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [91]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [92]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [93]:
station_name2 = 'Huairou'
year2 = 2015

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [94]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [95]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [96]:
station_name2 = 'Nongzhanguan'
year2 = 2016

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [97]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [98]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [99]:
station_name2 = 'Shunyi'
year2 = 2013

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [100]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [101]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [102]:
station_name2 = 'Tiantan'
year2 = 2014

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [103]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()


Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [104]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [105]:
station_name2 = 'Wanliu'
year2 = 2015

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [106]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [107]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

Menentukan Stasiun dan Tahun yang ingin di highlight

In [108]:
station_name2 = 'Wanshouxigong'
year2 = 2016

Memfilter dan menghitung rata-rata PM10 berdasarkan musim untuk data tertentu

In [109]:
data_filtered_musim = data_grouped[(data_grouped['year'] == year2) & (data_grouped['station'] == station_name2)]
rata_rata_musim = data_filtered_musim.groupby(['station', 'musim'])['PM10'].mean().reset_index()

Membuat dan menampilkan hasil visualisasi rata-rata konsentrasi PM10 per musim

In [110]:
fig = px.bar(rata_rata_musim,
             x='musim',
             y='PM10',
             title=f'Rata-rata PM10 per Musim Tahun {year2} di stasiun {station_name2}',
             labels={'musim': 'Musim', 'PM10': 'Rata-rata PM10'},
             color='musim')

fig.show()

**Insight:**
- Dari grafik garis yang menunjukkan rata-rata PM10 setiap bulan contohnya di stasiun Aotizhongxin pada tahun 2013, kita dapat mengamati pola tahunan dalam konsentrasi PM10.
- Dengan menebalkan garis untuk tahun tertentu, kita dapat dengan mudah membandingkan tahun tersebut dengan tahun-tahun lainnya.
- Dari grafik batang yang menunjukkan rata-rata PM10 per musim contohnya di stasiun Wanshouxigong pada tahun 2016, kita dapat melihat bagaimana konsentrasi PM10 bervariasi di antara musim


## Analisis Lanjutan (Opsional)

### Geospatial Analysis

Membuat DataFrame dengan menghilangkan data NaN pada kolom kualitas_udara.

In [111]:
AQI_df = AQI_all_df.dropna(subset=['kualitas_udara'])
AQI_df

station  latitude  longitude  year   aqi_pm10 kualitas_udara  month  \
0    Aotizhongxin   39.9092   116.3972  2013  79.655928         Sedang    NaN   
1    Aotizhongxin   39.9092   116.3972  2014  84.033077         Sedang    NaN   
2    Aotizhongxin   39.9092   116.3972  2015  79.262817         Sedang    NaN   
3    Aotizhongxin   39.9092   116.3972  2016  70.505994         Sedang    NaN   
4       Changping   40.2000   116.2000  2013  68.919711         Sedang    NaN   
5       Changping   40.2000   116.2000  2014  76.751244         Sedang    NaN   
6       Changping   40.2000   116.2000  2015  70.164366         Sedang    NaN   
7       Changping   40.2000   116.2000  2016  66.395176         Sedang    NaN   
8        Dingling   40.2350   116.1960  2013  62.747008         Sedang    NaN   
9        Dingling   40.2350   116.1960  2014  68.786120         Sedang    NaN   
10       Dingling   40.2350   116.1960  2015  66.112162         Sedang    NaN   
11       Dingling   40.2350   116.1960  2016  63.455943         Sedang    NaN   
12         Dongsi   39.9330   116.4160  2013  75.629594         Sedang    NaN   
13         Dongsi   39.9330   116.4160  2014  81.799801         Sedang    NaN   
14         Dongsi   39.9330   116.4160  2015  79.196621         Sedang    NaN   
15         Dongsi   39.9330   116.4160  2016  74.456523         Sedang    NaN   
16       Guanyuan   39.9510   116.3090  2013  77.095171         Sedang    NaN   
17       Guanyuan   39.9510   116.3090  2014  81.222532         Sedang    NaN   
18       Guanyuan   39.9510   116.3090  2015  76.423646         Sedang    NaN   
19       Guanyuan   39.9510   116.3090  2016  75.449343         Sedang    NaN   
20        Gucheng   39.9130   116.2040  2013  79.479608         Sedang    NaN   
21        Gucheng   39.9130   116.2040  2014  90.143810         Sedang    NaN   
22        Gucheng   39.9130   116.2040  2015  81.349313         Sedang    NaN   
23        Gucheng   39.9130   116.2040  2016  78.468812         Sedang    NaN   
24        Huairou   40.4300   116.6400  2013  71.202368         Sedang    NaN   
25        Huairou   40.4300   116.6400  2014  73.295830         Sedang    NaN   
26        Huairou   40.4300   116.6400  2015  68.070483         Sedang    NaN   
27        Huairou   40.4300   116.6400  2016  64.391575         Sedang    NaN   
28   Nongzhanguan   39.9320   116.4500  2013  75.157394         Sedang    NaN   
29   Nongzhanguan   39.9320   116.4500  2014  86.013504         Sedang    NaN   
30   Nongzhanguan   39.9320   116.4500  2015  77.024836         Sedang    NaN   
31   Nongzhanguan   39.9320   116.4500  2016  71.720666         Sedang    NaN   
32         Shunyi   40.1300   116.6520  2013  70.594343         Sedang    NaN   
33         Shunyi   40.1300   116.6520  2014  78.830288         Sedang    NaN   
34         Shunyi   40.1300   116.6520  2015  73.211100         Sedang    NaN   
35         Shunyi   40.1300   116.6520  2016  66.904821         Sedang    NaN   
36        Tiantan   39.8840   116.4140  2013  77.367569         Sedang    NaN   
37        Tiantan   39.8840   116.4140  2014  80.559271         Sedang    NaN   
38        Tiantan   39.8840   116.4140  2015  74.748370         Sedang    NaN   
39        Tiantan   39.8840   116.4140  2016  71.725261         Sedang    NaN   
40         Wanliu   39.9790   116.3070  2013  79.339980         Sedang    NaN   
41         Wanliu   39.9790   116.3070  2014  88.374769         Sedang    NaN   
42         Wanliu   39.9790   116.3070  2015  76.666635         Sedang    NaN   
43         Wanliu   39.9790   116.3070  2016  69.621214         Sedang    NaN   
44  Wanshouxigong   39.9050   116.3070  2013  78.856432         Sedang    NaN   
45  Wanshouxigong   39.9050   116.3070  2014  86.381655         Sedang    NaN   
46  Wanshouxigong   39.9050   116.3070  2015  78.784250         Sedang    NaN   
47  Wanshouxigong   39.9050   116.3070  2016  72.516236         Sedang    NaN   

    PM10 musim  
0    NaN   NaN  
1    N

Menentukan Stasiun dan Tahun yang ingin di highlight

In [112]:
nama_stasiun3 = 'Gucheng'
year3 = 2013

Menyaring data berdasarkan tahun dan stasiun serta mengambil lokasi geografis dari stasiun tersebut

In [113]:
data_filtered_p = AQI_df[AQI_df['year'] == year3]
station_data = AQI_df[AQI_df['station'] == nama_stasiun3].iloc[0]
location = [station_data['latitude'], station_data['longitude']]

Membuat peta

In [114]:
m = folium.Map(location=location, zoom_start=12,  tiles='CartoDB positron')

Menampilkan kualitas udara di berbagai stasiun pada peta dan menyediakan informasi tambahan melalui popup.

In [115]:
for index, row in data_filtered_p.iterrows():
    if row['kualitas_udara'] == 'Baik':
        color = 'green'
    elif row['kualitas_udara'] == 'Sedang':
        color = 'orange'
    else:  # 'Buruk'
        color = 'red'

    popup_text = f"Station: {row['station']}<br>Tahun: {row['year']}<br>Indeks Kualitas Udara: {row['kualitas_udara']}<br>Rata-Rata PM10/Tahun: {row['aqi_pm10']}"

    folium.Marker(
        location=[row['latitude'], row['longitude']],
        tooltip=row['station'],
        popup=folium.Popup(popup_text, max_width=200),
        icon=folium.Icon(color=color)
    ).add_to(m)

m

## Conclusion

- Dari hasil yang sudah divisualisasikan diatas, dapat diketahui bahwa Rata-Rata PM10 setiap bulan di semua stasiun dan setiap tahun hampir selalu naik pada bulan ke-9 dan mengalami penurunan pada bulan ke-5. Kemungkinan hal ini disebabkan oleh faktor-faktor lain yang bisa dianalisis lebih lanjut
- Dari hasil yang sudah divisualisasikan diatas, dapat diketahui bahwa Rata-Rata PM10 pada musim Semi di semua stasiun dan setiap tahun selalu mencapai angka 100 dan pada musim Panas sangat jarang yang mencapai angka 100. Sama seperti hasil visualisasi pertanyaan pertama kemungkinan hal ini disebabkan oleh faktor-faktor lain yang bisa dianalisis lebih lanjut